lib/Controller/ApiController.php

```diff
@@ -1179,6 +1180,85 @@ public function newSubmission(int $formId, array $answers, string $shareHash = '
 		return new DataResponse();
 	}
 
+	/**
+	 * Update an existing submission
+	 *
+	 * @param int $formId the form id
+	 * @param int $submissionId the submission id
+	 * @param array $answers [question_id => arrayOfString]
+	 * @param string $shareHash public share-hash -> Necessary to submit on public link-shares.
+	 * @return DataResponse
+	 * @throws OCSBadRequestException
+	 * @throws OCSForbiddenException
+	 */
+	#[CORS()]
+	#[NoAdminRequired()]
+	#[NoCSRFRequired()]
+	#[PublicPage()]
+	#[ApiRoute(verb: 'POST', url: Constants::API_BASE . 'forms/{formId}/submissions/{submissionId}', requirements: Constants::API_V3_REQUIREMENTS)]
```

Comment:
This should be a PUT request, as we're updating an existing entity.
```suggestion
	#[ApiRoute(verb: 'PUT', url: Constants::API_BASE . 'forms/{formId}/submissions/{submissionId}', requirements: Constants::API_V3_REQUIREMENTS)]
```

Reply:
@provokateurin or would a PATCH be better here?

Reply:
HTTP semantics wise PATCH is used for partial updates of a resource and PUT for updating the entire resource.
I don't have much context, but from the looks of the method this updates the entire submission, so PUT seems appropriate.